src/main/java/control/TimelineController.java

```diff
@@ -87,10 +106,93 @@ public void shotChangedHandler(CameraShotBlockUpdatedEvent event) {
         // Adjust model
         shot.setBeginCount(changedBlock.getBeginCount());
         shot.setEndCount(changedBlock.getEndCount());
-        // Remove shot from previous timeline and add to new one
-        previousTimeline.removeShot(shot);
-        this.project.getCameraTimelines()
-                .get(changedBlock.getTimetableNumber()).addShot(shot);
+
+        CameraTimeline newCameraTimeline = this.project.getCameraTimelines()
+                .get(changedBlock.getTimetableNumber());
+
+        // Remove shot from previous timeline and add to new one if changed
+        if (event.getOldTimelineNumber() != changedBlock.getTimetableNumber()) {
+            previousTimeline.removeShot(shot);
+            newCameraTimeline.addShot(shot);
+        }
+
+        // check for collisions
+        checkCollisions(changedBlock.getTimetableNumber(),
+                event.getOldTimelineNumber(), changedBlock);
+    }
+
+    /**
+     * Check collisions for new camerablock.
+     * @param timelineNumber - the timelinenumber where the camerashotblock is added
+     * @param cameraShotBlock - the camerashotblock to check collisions with
+     */
+    private void checkCollisions(int timelineNumber, CameraShotBlock cameraShotBlock) {
+        checkCollisions(timelineNumber, -1, cameraShotBlock);
+    }
+
+    /**
+     * Check for collisions (added and removed).
+     * @param timelineNumber - the current timeline number to check new collisions
+     * @param oldTimelineNumber - the old timeline number to check which collisions are not valid
+     *                          anymore should be negative when adding new block
+     * @param cameraShotBlock - the shotblock to check collisions with
+     */
+    private void checkCollisions(int timelineNumber, int oldTimelineNumber,
+                                 CameraShotBlock cameraShotBlock) {
+        // Get the correct timeline
+        CameraTimeline timeline = project.getCameraTimelines().get(timelineNumber);
+        // Remove collisions from shot if added to new timeline
+        if (oldTimelineNumber >= 0) {
+            removeCollisionFromCameraShotBlock(cameraShotBlock);
+            // Remove overlaps for non-overlapping shotblocks
+            ArrayList<CameraShotBlock> toRemove = new ArrayList<>();
+            this.overlappingCameraShotBlocks.stream().filter(shotBlock -> shotBlock.getShot()
```

Comment:
Please put each action in a stream on a new line. This greatly improves readability.

```java
List<String> strings = Arrays.stream(ppl)
        .map((x) -> x.toUpperCase())
        .filter((x) -> x.contains("(M)"))
        .collect(Collectors.toList());
```